src/pages/Home/Menu/MenuModal.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { MenuModalStyle } from "./MenuModal.style";
+import { MenuHandleStyle } from "./MenuHandle.style";
+import { CategoryLiStyle } from "./CategoryLi.style";
+import { CategoryPosition, HEADER_HEIGHT } from "../Home";
+import { SpringValue } from "react-spring";
+
+type Props = {
+  categories: CategoryPosition[];
+  activeCategory: string;
+  onOverlayClick: React.Dispatch<React.SetStateAction<boolean>>;
+  style: {
+    transform: SpringValue<string>;
+  };
+};
+
+export default function MenuModal({
+  categories,
+  activeCategory,
+  onOverlayClick,
+  style,
+}: Props) {
+  /*categoryClickHandler
+    click occurs
+    1. menuModal closed
+    2. scroll to the category position
+    3. Active category is automatically updated by scroll position useEffect.
+  */
+
+  const categoryClickHandler = (
+    e: React.MouseEvent<HTMLLIElement, MouseEvent>
+  ) => {
+    const newActiveCategoryName: string = e.currentTarget.innerHTML;
+    onOverlayClick(false);
+    const newActiveCategory = categories.find((category) => {
+      return category.name === newActiveCategoryName;
+    });
+    const activeOffsetY =
+      newActiveCategory !== undefined ? newActiveCategory.offsetY : 0;
+
+    window.scrollTo({ top: activeOffsetY - HEADER_HEIGHT, behavior: "smooth" });
+  };
+
+  return (
+    <MenuModalStyle style={style}>
+      <MenuHandleStyle />
+      <ul>
+        {categories.map((category) => {
+          return (
+            <CategoryLiStyle
+              key={category.name}
+              className={
+                category.name === activeCategory ? "activeCategory" : ""
```

Comment:
don't add classes like this
https://styled-components.com/docs/basics#adapting-based-on-props
Pass props to the styled component element

Reply:
Fixed https://github.com/Royleong31/restaurant-interface/commit/df96b35106f559f86d48e306ebeca572944a65f1#diff-ba697114f992ebc1481671cc48b357c4efd47987bce8e6deea44be9293459243R12
```
<CategoryLiStyle
      key={category.name}
      onClick={categoryClickHandler}
      color={category.name === activeCategory ? "#1c6dc9" : ""}
>
```
`color: ${(props) => props.color};`